index.ts

```diff
@@ -71,7 +72,27 @@ const SeverityAnnotationLevelMap = new Map<RuleSeverity, "warning" | "failure">(
     } else {
       const linter = new Linter(options);
 
-      const files = glob.sync(pattern!);
+      let files = glob.sync(pattern!);
+      if (onlyChanged) {
+        const pullRequest = ctx.payload.pull_request;
+        if (pullRequest) {
+          const response = await octokit.pulls.listFiles({
+            owner: ctx.repo.owner,
+            repo: ctx.repo.repo,
+            pull_number: pullRequest.number,
+          });
+          const changedFiles = response.data.map((f) => f.filename);
+          files = files.filter((f: string) => changedFiles.includes(f));
+        } else {
+          const response = await octokit.repos.getCommit({
```

Comment:
It won't get changed files correctly if user pushed multiple commits

Reply:
All you have is `ctx.sha` so this should work OK.